### DISCUSSION TITLE
To get a solution for reducing the time of installation process.

### DISCUSSION THREAD

**Discussion by Vaibhav1825** [oleg-shilo/wixsharp#1711] [Q&A]
Hello Oleg,
We have a setup where we are creating 15-20 web applications under Default Web Site. While doing this we are deploying files required for IIS at InstallDir location. The size of these files is around 400Mb and it has around 40000 files in it. When we are building it is taking around 25 minutes. Also , while running the setup it is taking 15-16 minutes.
Is there any way so that we can reduce this time?

**Comment by oleg-shilo** [comment]
Vaibhav, unfortunately, I have no answer for you.

WixSharp involvement is limited to building MSI and implementing custom actions. Thus if you know that the custom action is contributing to the time overhead then indeed you can look at WixSharp. But for everything else, it is all controlled by MSI.

If i5 mins is fundamentally unacceptable and represents a business risk for you then you can consider this hacky approach that tbh I cannot recommend unless your business depends on it:
- deploy all 400Mb to the destination in a single step.
- after the files are copied set them up programmatically to be web apps (e.g. configure virtual dirs) 
- on uninstall do it all in the opposite direction

I am not even sure it will be faster. However it has a chance as it will be only a single file copy operation.
